### ISSUE TITLE
Use poetry for bootstrap dependency mgmt

### ISSUE THREAD

**Issue by eloquence** [freedomofpress/securedrop-builder#468]
Once #467 lands, we can migrate the bootstrap dependency management from `pip-compile` to `poetry`.

**Comment by eloquence** [comment]
Took a first stab at this in e39d74065c299a37463a0e7372cbc7ee5329da99. Our reqs are pretty outdated so I had to version-pin a couple more to avoid incompatibilities -- however nothing should be _older_ than in the current bootstrap. I think a holistic refresh (#464) would be a good follow-up task.

Currently this is failing CI on Bookworm; as far as I can tell, it's to attempting a source instead of wheel install for `cython`, and then rightly complaining that the sha256 for the wheel doesn't match the source tarball. Not sure why it's doing that, see https://app.circleci.com/jobs/github/freedomofpress/securedrop-builder/30477 for CI output. Will try to repro with Python 3.11 and if necessary with a Bookworm VM.

**Comment by eloquence** [comment]
In hindsight it of course makes sense that Python 3.11 wants a different version of Cython. I've built one and added it in f7465d074613d84bd87d70b55bfa27eeac24afb1, which make the bookworm jobs happy. Open Q: how much diff review do we want to apply to these core Python ecosystem dependencies? Will raise that in standup tomorrow. 

Otherwise next up: making reprotest happy.
